src/index.ts

```diff
@@ -1,10 +1,51 @@
 import process from 'process';
 import debounce from 'debounce';
 import chokidar from 'chokidar';
+import { promises as fs } from 'fs-extra';
 import { debug } from './logging';
 import { buildPath, restoreOldDirectories } from './build';
 import argv from './argv';
 
+async function getWatchPaths(modulePaths: string[]): Promise<string[]> {
+  const srcPaths: string[] = [];
+  for (let i = 0; i < modulePaths.length; i++) {
+    const path = modulePaths[i];
+    try {
+      // first look for an src/ dir
+      const srcPath = `${path}/src`;
+      const stats = await fs.stat(srcPath);
+      if (stats.isDirectory()) {
+        srcPaths.push(srcPath);
+      }
+      continue;
+    } catch {}
+
+    try {
+      // then look for a lib/ dir
+      const libPath = `${path}/lib`;
+      const stats = await fs.stat(libPath);
+      if (stats.isDirectory()) {
+        srcPaths.push(libPath);
+      }
+      continue;
+    } catch {}
+
+    try {
+      // then look for a lib/ dir
+      const libsPath = `${path}/libs`;
+      const stats = await fs.stat(libsPath);
+      if (stats.isDirectory()) {
+        srcPaths.push(libsPath);
+      }
+      continue;
+    } catch {}
+
+    // if none of the above dirs were found, watch root path
+    srcPaths.push(path);
```

Comment:
If you do that, you will probably have a infinite loop, as the build files are generally located in the project directory.
So it will : 

1. build your project in the `build/` dir
2. detect that the `build` dir has change
3. go to step one